packages/common/src/i18n/format_date.ts

```diff
@@ -83,6 +89,42 @@ export function formatDate(
   locale: string,
   timezone?: string,
 ): string {
+  if (ngDevMode) {
+    if (format.includes('Y') && !format.includes('w')) {
```

Comment:
A couple minor proposals:

- Add an extra check for the `ngDevMode` (i.e. `typeof ngDevMode === 'undefined'`), since the `formatDate` can be used in the code before ngDevMode gets fully initialized (for cases when a project is build without Angular CLI)
- I think we can move all the logic to a separate function, to keep the `formatDate` function itself more readable

```suggestion
  if (typeof ngDevMode === 'undefined' || ngDevMode) {
    assertValidDateFormat(format);
```